lib/puppet/pops/issues.rb

```diff
@@ -58,7 +58,8 @@ def format(hash ={})
   # in the issue's message producing block.
   # @api private
   #
-  class MessageData
+  class MessageData < Puppet::Pops::LabelProvider
```

Comment:
Regarding delegation. Delegating will not make a future I18N transition easier because the methods themselves are wrong. `a_an`, `a_an_uc` for instance, is hardly applicable in, say Swedish. We would need others though, to differentiate 'en', 'ett' etc. Hence, all issues and the label provider must be rewritten.

The only way to make it simpler is to retain my previous attempt for this PR and go via `provider.a_an` in the actual text of the message since that's the place where things must change.